modules/programs/nushell.nix

```
@@ -145,11 +145,11 @@ in {
     };
 
     environmentVariables = mkOption {
-      type = types.attrsOf types.str;
+      type = types.attrsOf (pkgs.formats.json { }).type;
       default = { };
       example = { FOO = "BAR"; };
```

I think you could add something like, so that it's obvious that more than just a string is supported:
```suggestion
      example = {
        FOO = "BAR";
        BOOLEAN_VAR = true;
        NUMERIC_VAR = 4;
        LIST_VAR = [ "elem1" 2 ];
      };
```

(could also be something more descriptive/real-world-use)